### PR TITLE
Make some fixes

### PR DESCRIPTION
**Pull request by ManyTheFish**
#### Fix original lengths method

original_lengths method was returning the char index instead of the char length (char_index + 1) and the byte index was returning the normalized length instead of the original one

#### Refactor script detection

split the text by script in order to use the right specialized pipeline when several scripts are used in the same text like "Ipad 包"

### CODE REVIEW

**Comment by ManyTheFish** [comment]
poke @Kerollmops

**Comment by ManyTheFish** [comment]
poke @Kerollmops

File: src/classifier/classifier.rs [hunk 1]
```diff
@@ -53,6 +53,22 @@ where
     }
 }
 
+fn classify_separator(c: char) -> Option<SeparatorKind> {
+    match deunicode_char(c)?.chars().next()? {
+        // Prevent deunicoding cyrillic chars (e.g. ь -> ' is incorrect)
+        _ if ('\u{0410}'..='\u{044f}').contains(&c) => None, // russian cyrillic letters [а-яА-Я]
+        _ if c == '\u{00a0}' => None,                        // non-breaking space
```
Comment (Kerollmops) [inline src/classifier/classifier.rs:60]: I would prefer that we consider this non-breaking space a real space for the engine to match more things.
https://en.wikipedia.org/wiki/Non-breaking_space

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Review by Kerollmops — DISMISSED** [review]

**Review by Kerollmops — DISMISSED** [review]

**Review by Kerollmops — APPROVED** [review]
Allez, c'est bon!

**Comment by ManyTheFish** [comment]
bors merge

**Pull request merged**